pkg/assembler/backends/arangodb/backend.go

```diff
@@ -652,14 +466,225 @@ func createGraph(ctx context.Context, db driver.Database, graphName string, edge
 
 func createMissingEdgeCollection(ctx context.Context, graph driver.Graph, edgeDefinitions []driver.EdgeDefinition) error {
 	fmt.Println("Checking for collection to exists ")
-	for _, edgeDefination := range edgeDefinitions {
-		exists, _ := graph.EdgeCollectionExists(ctx, edgeDefination.Collection)
+	for _, edgeDefinition := range edgeDefinitions {
+		// check if the edge collection itself exists
+		exists, _ := graph.EdgeCollectionExists(ctx, edgeDefinition.Collection)
 		if !exists {
-			_, err := graph.CreateEdgeCollection(ctx, edgeDefination.Collection, driver.VertexConstraints{From: edgeDefination.From, To: edgeDefination.To})
+			_, err := graph.CreateEdgeCollection(ctx, edgeDefinition.Collection, driver.VertexConstraints{From: edgeDefinition.From, To: edgeDefinition.To})
 			if err != nil {
 				return err
 			}
 		}
+
+		// Note:: even though the edge collection exists, its corresponding vertex collections might not
+		var vertexCollections []string
+		vertexCollections = append(vertexCollections, edgeDefinition.From...)
+		vertexCollections = append(vertexCollections, edgeDefinition.To...)
+
+		// create the edge collection if any vertex collection is missing
+		for _, vertexCollection := range vertexCollections {
+			exists, err := graph.VertexCollectionExists(ctx, vertexCollection)
+			if err != nil {
+				return err
+			}
+
+			if !exists {
+				_, err := graph.CreateVertexCollection(ctx, vertexCollection)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getCollectionIndexMap() map[string][]index {
+	collectionIndexMap := make(map[string][]index)
+
+	collectionIndexMap[artifactsStr] = []index{
+		*(initIndex("byDigest", []string{"digest"}, true)),
```

Comment:
is there a reason why you are taking a reference only to de-reference it for the map? Should change to:

```
func initIndex(name string, fields []string, unique bool) index {
	return index{
		name: name,
		fields: fields,
		unique: unique,
	}
}
```